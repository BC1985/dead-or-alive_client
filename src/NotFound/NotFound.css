.content {
  margin-bottom: 50px;
}
.not-found-container {
  text-align: center;
  font-family: "Francois One", sans-serif;
  color: gray;
}
.fact-card {
  background-color: orangered;
  padding: 20px;
  color: white;
  border-radius: 10px;
  text-align: left;
}
/* ----------- iPhone 5, 5S, 5C and 5SE ----------- */

/* Portrait and Landscape */
@media only screen and (min-device-width: 320px) and (max-device-width: 568px) and (-webkit-min-device-pixel-ratio: 2) {
  .not-found-container {
    font-size: 1.2em;
  }

  .fact-card {
    margin: 30px auto;
  }
  .error-back-button {
    width: 200px;
    padding: 10px;
  }
}
/* Portrait */
@media only screen and (min-device-width: 320px) and (max-device-width: 568px) and (-webkit-min-device-pixel-ratio: 2) and (orientation: portrait) {
  .not-found-container {
    /* font-size: 1.2em; */
    padding: 40px;
  }
  #error-graphic {
    width: 200px;
    margin: 30px 0px;
  }
  .error-back-button {
    width: 100%;
    padding: 10px;
  }
}

/* Landscape */
@media only screen and (min-device-width: 320px) and (max-device-width: 568px) and (-webkit-min-device-pixel-ratio: 2) and (orientation: landscape) {
  .not-found-container {
    padding: 50px;
  }
  #error-graphic {
    width: 50%;
    /* padding: 30px; */
  }
}

/* ----------- iPhone 6, 6S, 7 and 8 ----------- */

/* Portrait and Landscape */
@media only screen and (min-device-width: 375px) and (max-device-width: 667px) and (-webkit-min-device-pixel-ratio: 2) {
  .not-found-container {
    text-align: center;
    font-family: "Francois One", sans-serif;
    color: gray;
  }
}
/* Portrait */
@media only screen and (min-device-width: 375px) and (max-device-width: 667px) and (-webkit-min-device-pixel-ratio: 2) and (orientation: portrait) {
  .not-found-container {
    padding: 50px;
  }
  .error-back-button {
    width: 100%;
    padding: 10px;
  }
}

/* Landscape */
@media only screen and (min-device-width: 375px) and (max-device-width: 667px) and (-webkit-min-device-pixel-ratio: 2) and (orientation: landscape) {
  .not-found-container {
    font-size: 1.2em;
    padding: 60px;
  }
  .fact-card {
    margin-bottom: 40px;
  }

  #error-graphic {
    width: 30%;
    margin: 30px;
  }
  .content {
    width: 100%;
  }
  .error-back-button {
    width: 200px;
    padding: 10px;
  }
}

/* ----------- iPhone 6+, 7+ and 8+ ----------- */
/* Portrait and Landscape */
@media only screen and (min-device-width: 414px) and (max-device-width: 736px) and (-webkit-min-device-pixel-ratio: 3) {
  .not-found-container {
    text-align: center;
    font-family: "Francois One", sans-serif;
    color: gray;
  }

  .error-back-button {
    width: 200px;
    padding: 10px;
  }
}
/* Portrait */
@media only screen and (min-device-width: 414px) and (max-device-width: 736px) and (-webkit-min-device-pixel-ratio: 3) and (orientation: portrait) {
  .error-back-button {
    width: 100%;
    padding: 10px;
  }
}

/* Landscape */
@media only screen and (min-device-width: 414px) and (max-device-width: 736px) and (-webkit-min-device-pixel-ratio: 3) and (orientation: landscape) {
  .not-found-container {
    font-size: 1.2em;
    padding: 40px;
  }
  #error-graphic {
    width: 40%;
    padding: 40px;
  }
  /* .content {
    margin-bottom: 30px;
  } */
}

/* ----------- iPhone X ----------- */

/* Portrait */
@media only screen and (min-device-width: 375px) and (max-device-width: 812px) and (-webkit-min-device-pixel-ratio: 3) and (orientation: portrait) {
}

/* Landscape */
@media only screen and (min-device-width: 375px) and (max-device-width: 812px) and (-webkit-min-device-pixel-ratio: 3) and (orientation: landscape) {
  .not-found-container {
    padding: 50px;
    font-size: 1.2em;
  }
  #error-graphic {
    width: 40%;
    padding: 40px;
  }
  .fact-card {
    margin: 40px 50px;
  }
  .content {
    margin: 0 50px;
  }
}

/* ----------- Galaxy S6 ----------- */

/* Portrait */
@media screen and (device-width: 360px) and (device-height: 640px) and (-webkit-device-pixel-ratio: 4) and (orientation: portrait) {
}

/* Landscape */
@media screen and (device-width: 360px) and (device-height: 640px) and (-webkit-device-pixel-ratio: 4) and (orientation: landscape) {
}

/* ----------- iPad 3, 4 and Pro 9.7" ----------- */
/* Portrait and Landscape */
@media only screen and (min-device-width: 768px) and (max-device-width: 1024px) and (-webkit-min-device-pixel-ratio: 2) {
  .not-found-container {
    padding: 50px 100px;
    font-size: 1.2em;
  }
  #error-graphic {
    width: 40%;
    padding: 40px;
  }
  .fact-card {
    margin-bottom: 50px;
  }
  .error-back-button {
    width: 200px;
    padding: 10px;
  }
}
/* Portrait */
@media only screen and (min-device-width: 768px) and (max-device-width: 1024px) and (orientation: portrait) and (-webkit-min-device-pixel-ratio: 2) {
}

/* Landscape */
@media only screen and (min-device-width: 768px) and (max-device-width: 1024px) and (orientation: landscape) and (-webkit-min-device-pixel-ratio: 2) {
}

/* ----------- iPad Pro 10.5" ----------- */

/* Portrait and Landscape */
@media only screen and (min-device-width: 834px) and (max-device-width: 1112px) and (-webkit-min-device-pixel-ratio: 2) {
}
/* Portrait */
/* Declare the same value for min- and max-width to avoid colliding with desktops */
@media only screen and (min-device-width: 834px) and (max-device-width: 834px) and (orientation: portrait) and (-webkit-min-device-pixel-ratio: 2) {
}

/* Landscape */
/* Declare the same value for min- and max-width to avoid colliding with desktops */
@media only screen and (min-device-width: 1112px) and (max-device-width: 1112px) and (orientation: landscape) and (-webkit-min-device-pixel-ratio: 2) {
}
/* ----------- Non-Retina Screens ----------- */
@media screen and (min-device-width: 1200px) and (max-device-width: 1600px) and (-webkit-min-device-pixel-ratio: 1) {
  .not-found-container {
    padding: 10% 20%;
  }
  #error-graphic {
    width: 40%;
    padding: 40px;
  }
  .content {
    font-size: 1.2em;
  }
  .fact-card {
    margin-bottom: 50px;
  }
}

/* ----------- Retina Screens ----------- */
@media screen and (min-device-width: 1200px) and (max-device-width: 1600px) and (-webkit-min-device-pixel-ratio: 2) and (min-resolution: 192dpi) {
}
/* ----------- Large Screens ----------- */
@media screen and (min-device-width: 1600px) {
  .not-found-container {
    padding: 5% 20%;
    font-size: 12px;
  }
  #error-graphic {
    width: 40%;
    padding: 40px;
  }

  .fact-card {
    margin-bottom: 50px;
  }
  .error-back-button {
    display: none;
  }
}
