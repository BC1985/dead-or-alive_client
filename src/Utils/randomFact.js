export function generateRandomFact() {
  const randomFacts = [
    "One in five country music songs refer to alcohol, one in three to tears and one in seven to “mama”.",
    "Not a single member of the Beatles could read or write music.",
    "The most expensive instrument in the world is the “Lady Blunt” Stradivarius violin, which sold in 2011 for $15.9 million.",
    "A 2007 study found that music, classical in particular, can help make plants grow faster.",
    "Barry Manilow didn’t write the song called “I Write the Songs”",
    "In 2008, researchers discovered that loud music makes customers in bars drink more alcohol in less time.",
    "Leo Fender, the founder of the iconic electric guitar and bass brand and Rock and Roll Hall of Fame inductee, never learned to play either instrument.",
    "In 2015, Canadian astronaut Chris Hadfield released the first ever album recorded entirely in space, named Space Sessions: Songs from a Tin Can.",
    "For every $1,000 of music sold, the average musician only makes about $23.40.",
    "The first (and only) band to play on all seven continents is Metallica.",
    "In “Bohemian Rhapsody”, Freddie Mercury plays on the very same piano used by Paul McCartney in “Hey Jude”.",
    "Music, along with painting, poetry, literature and architecture, was Olympic event from 1912 until 1948.",
    "Mozart sold more CDs than Beyoncé in 2016.",
    "When you listen to music, the brain releases the same feel-good hormone (dopamine) it does during sex and eating.",
    "In 1952, American composer John Cage created a song titled “4’44”. It consisted of four minutes and 33 seconds of silence.",
    "The longest song title ever is Hoagy Carmichael’s 1943 “I'm a Cranky Old Yank in a Clanky Old Tank on the Streets of Yokohama with my Honolulu Mama Doin' Those Beat-o, Beat-o Flat-On-My-Seat-o, Hirohito Blues.”",
    "There are more people in Monaco’s orchestra than in its army.",
    "Bryan Adams photographed The Queen for a Canadian postal stamp commemorating Her Majesty’s Golden Jubilee.",
    "In the US in 2005, vinyl record sales only reached $35 million. In 2015, sales had exploded to $416 million.",
    "Florida passed a law requiring toddlers in state-run schools to listen to classical music every day, and in 1998 the governor of Georgia budgeted $105,000 per year to provide every child born in Georgia with a tape or CD of classical music. ",
    "The song “Torn” by Natalie Imbruglia is actually a cover. The original is by Ednaswap.",
    "Actor Mel Brooks was a combat engineer during the Battle of the Bulge. When the Germans played propaganda recordings over loudspeakers, Brooks responded by setting up his own sound system and played music by Al Jolson, a Jewish musician",
    "In 2008, an Ohio man named Andrew Vactor was sentenced to listening to classical music by the likes of Bach, Beethoven and Chopin for 20 hours, after being caught for playing rap music too loud, as an alternative to paying a full fine of $150. He only lasted 15 minutes before leaving.",
    "The song “Sweet Child of Mine” was written in five minutes.",
    "In 2006, VH1 Classic ran fundraiser for Hurricane Katrina relief in which viewers could donate $25 to see the music video of their choice. One viewer donated $35, 000. The viewer could have selected an hour’s worth of music videos from the 1960s through the early 1990s. However, they chose something different for his allotted hour, requesting continuous playing of 99 Luftballons.",
    "On the one-year anniversary of the Curiosity rover’s landing on Mars, it played “Happy Birthday” to itself making “Happy Birthday” the first song and Curiosity the first device used to play music on a foreign planet.",
    `Serial killer Aileen Wuornos was obsessed with Natalie Merchant's "Carnival." She requested they play it at her funeral.`,
    `Sarah McLachlan's song "Angel" is about heroin addiction.`,
    "Eye movement in music reading is an extremely complex phenomenon that involves a number of unresolved issues in psychology. Despite some 30 studies in this area over the past 70 years, little is known about the underlying patterns of eye movement in music reading.",
    '"Zombie" by The Cranberries is about terrorism in Northern Ireland.',
    "In retail stores slow music is played to keep you shopping and spend more. In restaurants fast music is played to speed up turnaround.",

    "Beck’s last album was released only in sheet music form, for others to interpret. No cd’s, no digital download, just sheet music.",
    'Shel Silverstein wrote "A Boy Named Sue."',
    "Famed jazz guitarist Django Reinhardt was in Paris during the Nazi occupation. Despite being a gypsy, handicapped, a player of a banned music style, and a known friend of blacks and Jews, the Germans left him alone because some higher Nazi officers were huge fans of his music.",
    "Garth Brooks is one of the last holdout big-name musicians who still refuses to put his music on iTunes because he disagrees with Apple’s approach to selling music. He will make his back catalog of hits and his new music available for download, but only through his own website.",
    "A McDonald’s in Mount Annan, Sydney Austrailia has been playing classical music on purpose to force teenagers off of its premises at late night.",
    "Hitchcock didn’t want music in Psycho’s shower scene. He practically doubled composer Bernard Herrmann’s salary after seeing the scene with music",
    "Classical and heavy metal music both attracts listeners with similar personalities but dissimilar ages. Younger members apparently go for heavy metal, while their older counterparts prefer classical. However, both have the same basic motivation: to hear something dramatic and theatrical, a shared “love of the grandiose",
    "Hans Zimmer, the composer/producer for over 150 movies, including The Lion King, Gladiator, all Pirates of the Caribbean movies, Pearl Harbor, Rush, Inception, the Dark Knight, etc.; had only 2 weeks of formal music education (piano lessons as a child).",
    "Beethoven won a sight reading piano competition by turning his sheet music upside down, playing it, then improvising on one of its themes for thirty minutes. His opponent, Daniel Steibelt, never returned to challenge him again.",
    "Police in Belfast, Northern Ireland used music from ice cream van to calm angry teen rioters.",
    "There is a music piece called “As Slow as Possible” which started in 2001 and has a duration of 639 years. It can take years for a single note to be played.",
    "In 2008, Warner Music Group was collecting over $5,000 a day ($2 million a year) in royalties for public usage of the “Happy Birthday” song. They have been known to charge up to $10,000 for usage of the song in film. This is also why restaurants have their own birthday songs when you visit.",
    `The US military would play Enter Sandman for hours on end to aid in the interrogation and torture of prisoners. When asked about this, Metallica’s James Hetfield responded “We’ve been punishing our parents, our wives, our loved ones with this music forever. Why should the Iraqis be any different?”`,
    "When Beethoven went deaf, he composed his music by hacking the legs off his piano and sitting on the floor so he could feel the vibrations.",
    "A Sea Organ in Croatia continuously plays music powered by waves that flow through tubes located underneath a set of large marble steps."
  ];
  let randomFact = randomFacts[Math.floor(Math.random() * randomFacts.length)];
  return randomFact;
}
