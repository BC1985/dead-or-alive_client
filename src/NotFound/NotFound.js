import React, { Component } from "react";
import { generateRandomFact } from "../Utils/randomFact";
import { NavLink } from "react-router-dom";
import error from "./error.png";
import "./NotFound.css";
class NotFound extends Component {
  render() {
    return (
      <div className="not-found-container">
        <h1>Uh oh...</h1>
        <div className="error-graphic">
          <img src={error} alt="404 error" id="error-graphic" />
        </div>
        <p style={{ textAlign: "left" }}>
          The person you requested is unavailable. The name has been submitted,
          and their information will be uploaded soon. In the meantime, here is
          an interesting music fact...
        </p>
        <div className="fact-card">{generateRandomFact()}</div>
        <NavLink to="/">
          <button>Back</button>
        </NavLink>
      </div>
    );
  }
}

export default NotFound;
