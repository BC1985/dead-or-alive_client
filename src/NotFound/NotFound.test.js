import React from "react";
import NotFound from "./NotFound";
import ReactDOM from "react-dom";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { generateRandomFact } from "../Utils/randomFact";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <MemoryRouter>
      <NotFound generateRandomFact={generateRandomFact} />
    </MemoryRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it("renders the UI as expected", () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <NotFound generateRandomFact={generateRandomFact} />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
