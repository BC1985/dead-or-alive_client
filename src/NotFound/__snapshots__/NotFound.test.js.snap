// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="not-found-container"
>
  <h1>
    Uh oh...
  </h1>
  <div
    className="error-graphic"
  >
    <img
      alt="404 error"
      id="error-graphic"
      src="error.png"
    />
  </div>
  <div
    className="content"
  >
    <p
      style={
        Object {
          "textAlign": "left",
        }
      }
    >
       is unavailable. The name has been submitted, and their information will be uploaded soon. Here is what I can tell you in the meantime: 
       
      <br />
      And now, here is an interesting music fact...
    </p>
  </div>
  <div
    className="fact-card"
  >
    Serial killer Aileen Wuornos was obsessed with Natalie Merchant's "Carnival." She requested they play it at her funeral.
  </div>
  <a
    aria-current="page"
    className="active"
    href="/"
    onClick={[Function]}
    style={Object {}}
  >
    <button
      className="error-back-button"
    >
      Back
    </button>
  </a>
</div>
`;
